File: TUnit.TestProject/BasicTests.cs
namespace TUnit.TestProject;

[Category("Smoke")]
public class BasicTests
{
    [Test]
    public void SynchronousTest()
    {
        // Dummy method
    }

    [Test]
    public async Task AsynchronousTest()
    {
        await Task.CompletedTask;
    }

    [Test]
    public async ValueTask ValueTaskAsynchronousTest()
    {
        await new ValueTask();
    }
}

File: TUnit.TestProject.FSharp/BasicTests.fs
namespace TUnit.TestProject.FSharp

open System.Threading.Tasks
open TUnit.Core

// Equivalent of BasicTests.cs

type BasicTests() =
    [<Test>]
    member _.SynchronousTest() =
        () // Dummy method

    [<Test>]
    member _.AsynchronousTest() : Task =
        Task.CompletedTask

    [<Test>]
    member _.ValueTaskAsynchronousTest() : ValueTask =
        ValueTask()
